@here/harp-mapview/lib/text/TextElement.ts

```diff
@@ -291,8 +291,25 @@ export class RenderState {
             this.state === FadingState.FadingOut;
         return visible;
     }
+
+    /**
+     * Creates a clone of this object, uid is ignored.
+     */
+    clone() {
+        return new RenderState(
+            this.state,
+            this.value,
+            this.startTime,
+            this.opacity,
+            this.lastFrameNumber,
+            this.fadingTime
+        );
+    }
 }
 
+// Unique ID counter
+let TEXT_ELEMENT_UID = 0;
```

Comment:
It would be better to make it as private static property of TextElement class:

```js
class TextElement {
    private static counter: number = 0;
    readonly uid: number = ++TextElementCounter;
}
```